From e340eca95a329fa49523525c5ae43389263b31cd Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Thu, 21 Aug 2014 13:06:19 -0400
Subject: [ethernet] qlcnic: Fix number of rings when we fall back from msix to legacy

Message-id: <1408626422-3941-96-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88669
O-Subject: [PATCH RHEL 7.1 095/138] qlcnic: Fix number of rings when we fall back from msix to legacy.
Bugzilla: 1089352
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Rajesh Borundia <rajesh.borundia@qlogic.com>

Bug 1089352 (https://bugzilla.redhat.com/show_bug.cgi?id=1089352)

net commit id 42beb3f2836a6063ceb8134dbac0e32df1deea26

o Driver was not re-setting sds ring count to 1 after failing
   to allocate msi-x interrupts.

Signed-off-by: Rajesh Borundia <rajesh.borundia@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_hw.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_hw.c
index 1e7d9e3..5673774 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_hw.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_hw.c
@@ -345,6 +345,7 @@ int qlcnic_83xx_setup_intr(struct qlcnic_adapter *adapter)
    if (qlcnic_sriov_vf_check(adapter))
     return -EINVAL;
    num_msix = 1;
+   adapter->drv_sds_rings = QLCNIC_SINGLE_RING;
    adapter->drv_tx_rings = QLCNIC_SINGLE_RING;
   }
  }
diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
index c884fc0..3e6e112 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
@@ -822,6 +822,7 @@ static int qlcnic_82xx_setup_intr(struct qlcnic_adapter *adapter)
 
   if (!(adapter->flags & QLCNIC_MSIX_ENABLED)) {
    qlcnic_disable_multi_tx(adapter);
+   adapter->drv_sds_rings = QLCNIC_SINGLE_RING;
 
    err = qlcnic_enable_msi_legacy(adapter);
    if (err)
-- 
1.7.1